### PR TITLE
Fix appearing extra labels when 'keras_generator' function is called

### PR DESCRIPTION
**Pull request by RuslanIsrafilov**
Hi @mdbloice, I found Augmentator very useful tool, but I faced with distracting bug with `keras_generator` function, so my pull request is gonna fix it. The bug is following:

I have a training set which contains images of two classes, so there's two subfolders in the `train_128_path`, and i'm doing this
```Python
p = Augmentor.Pipeline(train_128_path)
...
generator = p.keras_generator(batch_size, image_data_format="channels_first") # It's ok
...
# Second call
generator = p.keras_generator(batch_size, image_data_format="channels_first") 
# It's not ok, it behaves like there's three labels 
# and generator returns label vectors of size three
```
After the first call it creates directory `output` relative to `train_128_path`, and on the second call  `keras_generator` treats `output` like new image class.

I found problem with the function 'Pipeline._populate', it calls `scan` and passes `abs_output_directory` which actually is not absolute, but `scan` expects an absolute path (problem with the line [195](https://github.com/mdbloice/Augmentor/blob/master/Augmentor/ImageUtilities.py#L195))
```Python
abs_output_directory = os.path.join(source_directory, output_directory)
...
self.augmentor_images, self.class_labels = scan(source_directory, abs_output_directory)
```
So I've fixed it in the `scan` function, by the way removed extra spaces.

### CODE REVIEW

**Comment by mdbloice** [comment]
Hi @RuslanIsrafilov I've been trying to reproduce the error you are speaking about, and I can't seem to be able to do it. Could you post a minimal working example? 

P.S. are you by chance creating another `Pipeline` object after you have created the first one (in the same directory)?

**Comment by RuslanIsrafilov** [comment]
For instance, I have directory structure like this:
```
├── data
│   ├── 1
│   ├── 2
├── train.py
```

Content of train.py
```Python
import Augmentor

p = Augmentor.Pipeline('data')
p.crop_random(probability=1, percentage_area=0.5)

g = p.keras_generator(1)
print('\nlabel = {}'.format(next(g)[1]))
```

After the first run it will print
```
python3 ./train.py
Initialised with 2 image(s) found.
Output directory set to data/output.
label = [[0 1]]
```

And it will create new directory `data/output`. No doubt, I can run the same one more time
```
python3 ./train.py
Initialised with 2 image(s) found.
Output directory set to data/output.
label = [[1 0 0]] # <- three labels???
```

**Comment by mdbloice** [comment]
Ok yes indeed, I see now. That's a good find @RuslanIsrafilov , thanks for putting the effort in. 
Actually all this time I didn't realise that in `_populate()` I've actually been sending the relative path all along. I'll fix that as well now that you've pointed it out. 
Thanks again!

**Comment by RuslanIsrafilov** [comment]
@mdbloice, from my point of view, unconditional creation of output directory is not the best design decision. Perhaps it should be created somewhere in the `sample()` method. Calling the constructor of `Pipeline`doent's mean what I'm really going to write images to disk.

**Comment by mdbloice** [comment]
Yeah, it's an artefact of back when the only option was to write images to disk. I'll keep it in mind and change it to write that folder only when `sample()` is called. Thanks for the feedback.
